src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php

```diff
@@ -155,6 +155,12 @@ private function getEffectiveSerializerGroups(array $options, string $resourceCl
         if (isset($options['serializer_groups'])) {
             $groups = (array) $options['serializer_groups'];
 
+            if (isset($options['deserializer_groups'])) {
```

Comment:
Couldn't `deserializer_groups` be set while `serializer_groups` aren't? To cover this case, shouldn't this test be outside ` if (isset($options['serializer_groups']))`?